src/transformers/models/character_bert/tokenization_character_bert.py

```diff
@@ -0,0 +1,979 @@
+            split_tokens = whitespace_tokenize(text)
+        return split_tokens
+
+    def convert_mlm_token_to_id(self, token):
+        """Converts a token (str) into an id using the MLM vocab."""
+        return self.mlm_vocab.get(token, self.mlm_vocab.get(self.unk_token))
+
+    def convert_mlm_id_to_token(self, index):
+        """Converts an index (integer) into a token (str) using the MLM vocab."""
+        return self.mlm_ids_to_tokens.get(index, self.unk_token)
+
+    # NOTE: the following two methods have misleading names since we are
+    # working with character id lists instead of standard token ids.
+    # Changing these names breaks a lot of things so we keep them for now.
+    def _convert_token_to_id(self, token):
+        """Converts a token (str) into a list of character ids (integer)."""
+        if token == self.cls_token:
+            character_ids = self.cls_character_ids
+        elif token == self.sep_token:
+            character_ids = self.sep_character_ids
+        elif token == self.mask_token:
+            character_ids = self.mask_character_ids
+        elif token == self.pad_token:
+            character_ids = self.pad_character_ids
+        else:
+            token_bytes = token.encode("utf-8", "ignore")[: (self.max_word_length - 2)]
+            character_ids = [self.pad_character_id] * self.max_word_length
+            character_ids[0] = self.bow_character_id
+            for k, index in enumerate(token_bytes, start=1):
+                character_ids[k] = index
+            character_ids[len(token_bytes) + 1] = self.eow_character_id
+            # NOTE: we shift everything so that the PAD token
+            #       can be assigned the all zeros vector
+            character_ids = [(index + 1) for index in character_ids]
+        return character_ids
+
+    def _convert_id_to_token(self, character_ids):
+        """Converts a list of character ids (integer) intp a token (str)."""
+        assert len(character_ids) == self.max_word_length, (
+            f"Got a character sequence of length {len(character_ids)} while "
+            f"`max_word_length={self.max_word_length}`"
+        )
+
+        if character_ids == self.cls_character_ids:
+            return self.cls_token
+        elif character_ids == self.sep_character_ids:
+            return self.sep_token
+        elif character_ids == self.mask_character_ids:
+            return self.mask_token
+        elif character_ids == self.pad_character_ids:
+            return self.pad_token
+        else:
+            character_ids_ = [(index - 1) for index in character_ids]
+            utf8_codes = list(
+                filter(
+                    lambda x: (
+                        (x != self.pad_character_id) and (x != self.bow_character_id) and (x != self.eow_character_id)
+                    ),
+                    character_ids_,
+                )
+            )
+            return bytes(utf8_codes).decode("utf-8")
+
+    def convert_ids_to_tokens(
```

Comment:
you should not have to overload this

Reply:
I think that I do, given the shape differences I mentioned below. The "id" is actually a list of ids here :/

Reply:
And if you were talking about the line `convert_ids_to_tokens` method, I also needed to overload it due to the first if/else statement not expecting the right shape, again due to the differences in how tokens are encoded 😩

Reply:
Alright. That where we might have a problem as the "API" is different. But as long as we document what is a token for this model (here a token is many classic token if I understood)